File: app/javascript/src/task/components/list_filters.js
import React from 'react';

import Link from 'src/route/containers/link';
import ModalLink from 'src/_common/containers/modal_link';

function TaskListFilters() {
  return (
    <div className='task-filters'>
      {'Filter: '}
      <Link to='tasks' baseClass='task-filter'>{'ALL'}</Link>
      <Link to='rootTasks' baseClass='task-filter'>{'ROOT'}</Link>
      <Link to='leafTasks' baseClass='task-filter'>{'LEAF'}</Link>
      <Link to='treeTasks' baseClass='task-filter'>{'TREE'}</Link>
      <ModalLink modalName='alpha' className='alpha-tag'>
        <sup>{'(alpha)'}</sup>
      </ModalLink>
    </div>
  );
}

export default TaskListFilters;

File: app/views/tasks/list_filters.rb
# frozen_string_literal: true

module Views
  module Tasks
    class ListFilters < Views::Base
      register_output_helper :active_link_to

      def view_template
        div(class: "task-filters") do
          plain("Filter:")
          filter_link("ALL", tasks_path)
          filter_link("ROOT", root_tasks_path)
          filter_link("LEAF", leaf_tasks_path)
          filter_link("TREE", tree_tasks_path)
          whitespace
          alpha_tag
        end
      end

      private

      def filter_link(label, path)
        whitespace
        active_link_to(label, path, class: "task-filter")
      end

      def alpha_tag
        span(class: "alpha-tag") do
          link_to(alpha_index_path, data: alpha_data) { sup { "(alpha)" } }
        end
      end

      def alpha_data
        { turbo: true, turbo_frame: "dialog" }
      end
    end
  end
end
